File: Medium/Number following a pattern/number-following-a-pattern.cpp
//{ Driver Code Starts
#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
class Solution{   
public:
    string printMinNumberForPattern(string S){
        // code here 
        string res;
        stack<int> st;
        for(int i=0;i<=S.length();i++)
        {
            st.push(i+1);
            if(S[i]=='I' || i==S.length())
            {
                while(!st.empty())
                {
                    res+=to_string(st.top());
                    res+="";
                    st.pop();
                }
            }
        }
        return res;
    }
};


//{ Driver Code Starts.
int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        string S;
        cin >> S;
        Solution ob;
        cout << ob.printMinNumberForPattern(S) << endl;
    }
    return 0; 
} 

// } Driver Code Ends

File: Medium/Number following a pattern/number-following-a-pattern.py
#User function Template for python3
class Solution:
    def printMinNumberForPattern(ob,S):
        t = []
        res=""
        for i in range(len(S)+1):
            t.append(i+1)
            if (i==len(S) or S[i]=="I"):
                while (len(t)>0):
                    res+=str(t[-1])
                    t.pop()

        return res
        


#{ 
 # Driver Code Starts
#Initial Template for Python 3
if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        
        S=str(input())

        ob = Solution()
        print(ob.printMinNumberForPattern(S))
# } Driver Code Ends